editor/utils/block-list.js

```diff
@@ -40,10 +41,19 @@ const INNER_BLOCK_LIST_CACHE = {};
 export function createInnerBlockList( uid, renderBlockMenu = noop ) {
 	if ( ! INNER_BLOCK_LIST_CACHE[ uid ] ) {
 		const InnerBlockListComponent = class extends Component {
-			componentWillReceiveProps( nextProps ) {
-				this.updateNestedSettings( {
-					supportedBlocks: nextProps.allowedBlocks,
-				} );
+			constructor( props ) {
+				super( props );
+				this.state = {};
+			}
+
+			static getDerivedStateFromProps( props ) {
+				const newSettings = {
+					allowedBlocks: props.allowedBlocks,
+					locking: props.locking === undefined ? props.parentLocking : props.locking,
+				};
+				if ( ! isShallowEqual( props.blockListSettings, newSettings ) ) {
+					props.updateNestedSettings( newSettings );
+				}
```

Comment:
On usage of `getDerivedStateFromProps`:
- [Strictly speaking](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops), it should return `null` instead of `undefined`.
- Side effects [are to be avoided](https://reactjs.org/blog/2018/05/23/react-v-16-4.html#1-avoid-side-effects-in-getderivedstatefromprops), and that is exactly what this particular method is doing. :)

Reply:
Hi @mcsf,
You are totally right it looks like getDerivedStateFromProps should be a pure function without side effects.
I updated the code to use componentDidUpdate as going back to componentWillReceive props is not a long-term option. componentDidUpdate also does not seems like a good way as we may have double render executions but it seems like the best option we have and according to the react docs users don't see the intermediate dom state.